cli/command/manifest/create_list.go

```diff
@@ -0,0 +1,82 @@
+package manifest
+
+import (
+	"fmt"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+
+	"github.com/docker/cli/cli"
+	"github.com/docker/cli/cli/command"
+	"github.com/docker/distribution/reference"
+	"github.com/docker/docker/registry"
+)
+
+type annotateOpts struct {
+	amend bool
+}
+
+func newCreateListCommand(dockerCli command.Cli) *cobra.Command {
+
+	opts := annotateOpts{}
+
+	cmd := &cobra.Command{
+		Use:   "create newRef manifest [manifest...]",
+		Short: "Create a local manifest list for annotating and pushing to a registry",
+		Args:  cli.RequiresMinArgs(2),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return createManifestList(dockerCli, args, opts)
```

Comment:
The pattern we use here is to handle `args` in this inline function so that we only have to pass `dockerCli, opts` to the other function.

`AnnotateOpts` should have a `target` and `manifests` field, which can be populated here from `args[0]` and `args[1:]`...

This pattern helps us keep the validation (the `Args:` line above) next to where we read  the array.